backend/src/main/java/com/huellapositiva/domain/actions/RegisterESALAction.java

```diff
@@ -33,9 +33,18 @@ public void execute(ESALRequestDto dto, String accountId) {
         if (contactPerson.hasESAL()) {
             throw new UserAlreadyHasESALException();
         }
-
-        Id id = esalRepository.newId();
-        ESAL esal = new ESAL(dto.getName(), id, contactPerson.getEmailAddress());
+        ESAL esal = ESAL.builder()
+                .id(esalRepository.newId())
```

Comment:
Isn't Id.newId() enough?

Reply:
You are right, there are still many places that use the esalRepository method or UUID.randomUUID and we should use Id.newId() everywhere.